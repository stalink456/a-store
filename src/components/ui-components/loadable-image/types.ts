export type LoadableImageType = {
  src: string;
  alt: string;
};
