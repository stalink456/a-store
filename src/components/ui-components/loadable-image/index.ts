export { LoadableImage } from './loadable-image';
