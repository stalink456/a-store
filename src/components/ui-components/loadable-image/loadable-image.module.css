@import "../../../css/variables.css";
@import "../../../css/animation.css";

img {
  width: 400px;
  height: 600px;
}

.loaded {
  position: relative;
  composes: opacity from global;
}
