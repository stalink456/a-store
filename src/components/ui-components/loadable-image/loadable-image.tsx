import React from 'react';
import { Skeleton } from '@alfalab/core-components/skeleton';
import { LoadableImageType } from './types';

import styles from './loadable-image.module.css';

export const LoadableImage: React.FC<LoadableImageType> = ({ src, alt }) => {
  const [isLoaded, setIsLoaded] = React.useState(false);

  const onImageLoaded = () => setIsLoaded(true);

  const onImageError = (e: React.ChangeEvent<HTMLImageElement>) => {
    e.target.src = 'images/image-not-found.jpeg';
  };

  return (
    <React.Fragment>
      <Skeleton visible={!isLoaded} animate={true}>
        <img
          className={`${isLoaded ? styles.loaded : ''}`}
          src={src}
          alt={alt}
          onLoad={onImageLoaded}
          onError={onImageError}
        />
      </Skeleton>
    </React.Fragment>
  );
};
