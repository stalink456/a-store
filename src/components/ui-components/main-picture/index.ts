export { MainPicture } from "./main-picture";
