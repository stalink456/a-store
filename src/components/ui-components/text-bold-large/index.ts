export { TextBoldLarge } from "./text-bold-large";
