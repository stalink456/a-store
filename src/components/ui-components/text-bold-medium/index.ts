export { TextBoldMedium } from "./text-bold-medium";
