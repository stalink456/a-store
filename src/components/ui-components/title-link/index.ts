export { TitleLink } from './title-link';
