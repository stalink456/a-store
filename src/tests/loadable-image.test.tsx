import React from "react";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import { LoadableImage } from "components/ui-components/loadable-image";
import * as image from "./mocks/get-image-200.json";
import * as imageNotFound from "./mocks/get-image-500.json";

const mockedImage = image["image"];
const mockedAlt = image["alt"];
const mockedNotFoundImage = imageNotFound["image"];
const mockedAltNotFoundImage = imageNotFound["alt"];

describe("LoadableImage component", () => {
  it("should render LoadableImage component", () => {
    const { baseElement } = render(
      <LoadableImage src={mockedImage} alt={mockedAlt} />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("shouldn't render LoadableImage component with non-existent image", async () => {
    render(
      <LoadableImage src={mockedNotFoundImage} alt={mockedAltNotFoundImage} />
    );

    const image = (await screen.findByRole("img")) as HTMLImageElement;

    fireEvent.error(image, {
      target: image,
    });

    await waitFor(() => {
      expect(image.src).toContain("images/image-not-found.jpeg");
    });
  });
});
