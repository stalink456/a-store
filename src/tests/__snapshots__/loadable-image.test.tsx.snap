// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadableImage component should render LoadableImage component 1`] = `
<body>
  <div>
    <div
      class="skeleton__component_1bshy skeleton__animate_1bshy"
    >
      <img
        alt="Футболка с бархатными стикерами"
        class=""
        src="http://qa-games.ru/astore/public/images/61646585.png"
      />
    </div>
  </div>
</body>
`;
